### PR TITLE
Merge repeating logs

### PR DESCRIPTION
**Pull request by Tristan-WorkGH**
Merge verbose logs with lots of repeating lines.

### CODE REVIEW

File: src/main/java/org/gridsuite/shortcircuit/server/service/ReportMapper.java [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+     *
+     * @implNote we assume there will always be at least one modification
+     */
+    protected ReporterModel forShortCircuitAnalysis(@NonNull final ReporterModel reporterModel) {
+        final ReporterModel newReporter = new ReporterModel(reporterModel.getTaskKey(), reporterModel.getDefaultName(), reporterModel.getTaskValues());
+        reporterModel.getReports().forEach(newReporter::report);
+        reporterModel.getSubReporters().forEach(reporter -> newReporter.addSubReporter(
+            switch (reporter.getTaskKey()) {
+                case "generatorConversion" -> forGeneratorConversion(reporter);
+                case "courcirc" -> forCourcirc(reporter);
```
Comment (jonenst) [inline src/main/java/org/gridsuite/shortcircuit/server/service/ReportMapper.java:102]: @geofjamg is this acceptable ? same for "forCourcirc" method ?

File: src/main/java/org/gridsuite/shortcircuit/server/service/ReportMapper.java [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+            log.trace("Unrecognized Reporter of type {}", reporter.getClass().getSimpleName());
+            return reporter;
+        }
+    }
+
+    /**
+     * Starting point, determine the root type
+     */
+    protected Reporter modifyReporterModel(@NonNull final ReporterModel reporterModel) {
+        if (reporterModel.getTaskKey().matches("^[a-fA-F\\d]{8}-[a-fA-F\\d]{4}-[a-fA-F\\d]{4}-[a-fA-F\\d]{4}-[a-fA-F\\d]{12}@ShortCircuitAnalysis$")) {
```
Comment (EstherDarkish) [inline src/main/java/org/gridsuite/shortcircuit/server/service/ReportMapper.java:72]: replace \\\\d by 0-9 ?

File: src/main/java/org/gridsuite/shortcircuit/server/service/ReportMapper.java (hunk 1 above)
Comment (Tristan-WorkGH) [reply src/main/java/org/gridsuite/shortcircuit/server/service/ReportMapper.java:108]: I have else separate with an interface in [another branch](/gridsuite/shortcircuit-server/tree/feature/concat_courcic_logs), so we can keep the ShortCircuitReports class in our package.

File: src/test/java/org/gridsuite/shortcircuit/server/assertj/AbstractReportAssert.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
+package org.gridsuite.shortcircuit.server.assertj;
```
[the remark this answers is not in the document]
Comment (Tristan-WorkGH) [reply src/test/java/org/gridsuite/shortcircuit/server/assertj/AbstractReportAssert.java:1]: check and done

File: src/main/java/org/gridsuite/shortcircuit/server/service/ReportMapper.java (hunk 2 above)
Comment (Tristan-WorkGH) [reply src/main/java/org/gridsuite/shortcircuit/server/service/ReportMapper.java:72]: not convinced, but oh well

File: src/main/java/org/gridsuite/shortcircuit/server/service/ReportMapper.java [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+            final Matcher matcherTransientReactanceTooLow = patternTransientReactanceTooLow.matcher(report.getDefaultMessage());
+            if (matcherTransientReactanceTooLow.matches()) { //we match line "X.ABCDEF1 : transient reactance too low ==> generator ignored"
+                logsTransientReactanceTooLow.add(matcherTransientReactanceTooLow.group(1));
+            } else { //we keep this log as it
+                newReporter.report(report);
+            }
+        }
+        /* finalize computation */
+        log.debug("Found {} lines in courcirc logs matching \"MYNODE : transient reactance too low ==> generator ignored\"", logsTransientReactanceTooLow.size());
+        newReporter.report("TransientReactanceTooLow", "${nb} node(s) with transient reactance too low ==> generator ignored\n${nodes}",
```
Comment (jonenst) [inline src/main/java/org/gridsuite/shortcircuit/server/service/ReportMapper.java:145]: don't add report if size is 0 ?

File: src/main/java/org/gridsuite/shortcircuit/server/service/ReportMapper.java [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+        final ReporterModel newReporter = new ReporterModel(reporterModel.getTaskKey(), reporterModel.getDefaultName(), reporterModel.getTaskValues());
+        reporterModel.getSubReporters().forEach(newReporter::addSubReporter);
+        /* preparing */
+        final Pattern patternTransientReactanceTooLow = Pattern.compile("^(.+) +: transient reactance too low ==> generator ignored$", Pattern.CASE_INSENSITIVE);
+        final List<String> logsTransientReactanceTooLow = new ArrayList<>(newReporter.getReports().size());
+        /* analyze and compute logs in one pass */
+        for (final Report report : reporterModel.getReports()) { //we modify logs conditionally here
+            final Matcher matcherTransientReactanceTooLow = patternTransientReactanceTooLow.matcher(report.getDefaultMessage());
+            if (matcherTransientReactanceTooLow.matches()) { //we match line "X.ABCDEF1 : transient reactance too low ==> generator ignored"
+                logsTransientReactanceTooLow.add(matcherTransientReactanceTooLow.group(1));
```
Comment (jonenst) [inline src/main/java/org/gridsuite/shortcircuit/server/service/ReportMapper.java:138]: @SlimaneAmar do we want to keep the old reports, and just add one summary with a different verbosity ? Or remove the original reports entirely ?

File: src/test/java/org/gridsuite/shortcircuit/server/ReporterLogsTest.java [hunk 6]
```diff
@@ -0,0 +1,246 @@
...
+            PlatformConfig.setDefaultConfig(new TestPlatformConfigProvider().getPlatformConfig()); //PowsyblException: Multiple platform configuration providers found
+        } catch (final UncheckedIOException ex) {
+            //there is an error with files on CI
+            final Path cfgDir = Paths.get(".", "unittests").toAbsolutePath();
+            PlatformConfig.setDefaultConfig(new PlatformConfig(PlatformConfig.loadModuleRepository(cfgDir, "config"), cfgDir));
+        }
+
+        rootReporter = new ReporterModel(ROOT_REPORTER_ID, ROOT_REPORTER_ID);
+        final Reporter reporter = rootReporter.createSubReporter(SHORTCIRCUIT_TYPE_REPORT, SHORTCIRCUIT_TYPE_REPORT + " (${providerToUse})", "providerToUse", "Courcirc");
+        //assertThat(reporter).isInstanceOf(ReporterModel.class);
```
Comment (jonenst) [inline src/test/java/org/gridsuite/shortcircuit/server/ReporterLogsTest.java:79]: why leave commented code

Comment (Tristan-WorkGH) [reply src/test/java/org/gridsuite/shortcircuit/server/ReporterLogsTest.java:79]: oupsy

File: src/main/java/org/gridsuite/shortcircuit/server/service/ReportMapper.java (hunk 5 above)
Comment (SlimaneAmar) [reply src/main/java/org/gridsuite/shortcircuit/server/service/ReportMapper.java:138]: I think we should ask Pauline

File: src/main/java/org/gridsuite/shortcircuit/server/service/ShortCircuitReportMapper.java [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+        }
+    }
+
+    /**
+     * Starting point, determine the root type
+     */
+    private Reporter modifyReporterModel(@NonNull final ReporterModel reporterModel) {
+        if (reporterModel.getTaskKey().matches("^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}@ShortCircuitAnalysis$")) {
+            log.debug("ShortCircuitAnalysis root node found, will modify it!");
+            return forUuidAtShortCircuitAnalysis(reporterModel);
```
Comment (jonenst) [inline src/main/java/org/gridsuite/shortcircuit/server/service/ShortCircuitReportMapper.java:74]: only clone if it's actually courcic. providerToUse value == "Courcirc"

**Review by jonenst — APPROVED** [review]

**Review by EstherDarkish — APPROVED** [review]
Code OK, Tests OK

**Review by SlimaneAmar — APPROVED** [review]

**Pull request merged**